client/my-sites/email/inbox/mailbox-selection-list/index.jsx

```diff
@@ -93,6 +94,10 @@ MailboxItemIcon.propType = {
 };
 
 const MailboxItem = ( { mailbox } ) => {
+	const siteId = useSelector( getSelectedSiteId );
+	const domains = useSiteDomains( siteId ?? null );
+	const firstTitanDomain = domains.find( hasTitanMailWithUs );
```

Comment:
Thought: it may be worth pushing this logic into a reusable hook like `useFirstTitanDomainForCurrentSite()` -- we have this pattern in two places.

Reply:
I like the idea of the reusable hook. I kinda pondered that at first as well.